LiteCore/Query/QueryParser.cc

```diff
@@ -719,14 +727,17 @@ namespace litecore {
             // Come up with a column title if there is no 'AS':
             if (title.empty()) {
                 if (result->type() == kString) {
-                    title = columnTitleFromProperty(Path(result->asString()), _propertiesUseAliases);
+                    title = columnTitleFromProperty(Path(result->asString()), _propertiesUseSourcePrefix);
                 } else if (result->type() == kArray && expr[0]->asString().hasPrefix('.')) {
-                    title = columnTitleFromProperty(propertyFromNode(result), _propertiesUseAliases);
+                    title = columnTitleFromProperty(propertyFromNode(result), _propertiesUseSourcePrefix);
                 } else {
                     title = format("$%u", ++anonCount); // default for non-properties
                 }
                 if (title.empty())
                     title = "*";        // for the property ".", i.e. the entire doc
+            } else {
+                _sql << " AS ";
+                writeSQLString(title);
```

Comment:
This is going to write a string, with single-quotes; does SQLite even allow that? To get double-quotes, add a double-quote character as a 2nd parameter.